src/Products/PythonScripts/PythonScript.py

```diff
@@ -261,6 +273,7 @@ def _compile(self):
                                fc.co_argcount)
         self.Python_magic = Python_magic
         self.Script_magic = Script_magic
+        linecache.clearcache()
```

Comment:
this empties **all** the cache, which will cause the next use of linecache to take a few milliseconds. In practice it's also possible to only remove the corresponding entry from the cache, like this:

```py
        linecache.cache.pop(
            getattr(self, '_filepath', None) or self.get_filepath(), None)
```

but it's not a documented the API and this is only called when python scripts are edited, so I thought it was better to just clear all the cache.